dev/io.openliberty.io.smallrye.config/resources/io/openliberty/io/smallrye/config/resources/SmallryeConfigMessages.nlsprops

```diff
@@ -0,0 +1,26 @@
+###############################################################################
+# Copyright (c) 2022 IBM Corporation and others.
+# All rights reserved. This program and the accompanying materials
+# are made available under the terms of the Eclipse Public License v1.0
+# which accompanies this distribution, and is available at
+# http://www.eclipse.org/legal/epl-v10.html
+#
+# Contributors:
+#     IBM Corporation - initial API and implementation
+###############################################################################
+#CMVCPATHNAME io.openliberty.io.smallrye.config/resources/io/openliberty/io/smallrye/config/resources/SmallryeConfigMessages.nlsprops
+#ISMESSAGEFILE TRUE
+#NLS_ENCODING=UNICODE
+
+#COMPONENTPREFIX CWWKC                 
+#COMPONENTNAMEFOR CWWKC Smallrye Config
+
+# Message numbers CWWKC0651-700 are reserved for this file.
+# NLS_MESSAGEFORMAT_VAR                
+# MSG_DESCRIPTIVE_NAME_CWWKC0000=CWWKC0000I: This is a message with inserts {0}
+# MSG_DESCRIPTIVE_NAME_CWWKC0000.explanation=Explanation text for the message
+# MSG_DESCRIPTIVE_NAME_CWWKC0000.useraction=User action text for the message
+
+WARNING_UPDATED_CONFIG_PROPERTY_NOT_USED_CWWKC0651=CWWKC0651W: The MicroProfile configuration value for the key {0} has changed since the checkpoint action completed on the server. The updated value might not be used by the application.
+WARNING_UPDATED_CONFIG_PROPERTY_NOT_USED_CWWKC0651.explanation=The configuration value was accessed during a checkpoint action and updated before the server process got restored. Depending on how the application accesses the configuration the updated value might not be used.
```

Comment:
"Depending on how the application accesses the configuration the updated value might not be used."
Can we have an example for this use case? If yes, please provide at least one example to include in the message!

Reply:
The example message will look like this:

```
[WARNING ] CWWKC0651W: The MicroProfile configuration value for the key test_key has changed since the checkpoint action completed on the server. The updated value might not be used by the application.
```

The example application class that would trigger such a message would look like this:

```java
package mpapp1;

import static org.junit.Assert.assertEquals;

import javax.enterprise.context.ApplicationScoped;
import javax.enterprise.context.Initialized;
import javax.enterprise.event.Observes;
import javax.inject.Inject;

import org.eclipse.microprofile.config.inject.ConfigProperty;

@ApplicationScoped
public class ApplicationScopedOnCheckpointBean {

    @Inject
    @ConfigProperty(name = "test_key", defaultValue = "annoValue")
    String testKey;

    public void applicationScopedValueTest() {
        // The value of test_key was not updated here because the property was accessed during the checkpoint.
        // A warning CWWKC0651W is thrown to let the user know that the updated value of the property might not be used during restore.
        check("defaultValue");
    }

    public void observeInit(@Observes @Initialized(ApplicationScoped.class) Object event) {
        System.out.println(getClass() + ": " + "Initializing application context");
        check("defaultValue");
    }

    private void check(String expected) {
        assertEquals("Wrong value for test key.", expected, testKey);
    }
}
```

Here the warning is making reference to the `test_key` that gets used by the field `testKey` that gets injected as a configuration property.  This particular application class has an `@ApplicationScope` which allows it to be created and injected before a checkpoint happens.  Here we are forcing that by having the bean observe the initialization of the ApplicationScope which forces CDI to aggressively create this bean as the application is starting.  When this happens the `testKey` will get injected with the current value set in config at the time the checkpoint occurs.  Now we can then change the value of the configuration for the `test_key` and restore the server.  At this point the server process will be restored, but this application scope bean instance will not get the updated value.

Thinking through that makes me think the message can be improved by indicating that we detected that the `test_key` was used and the value was read by the application before the checkpoint.  Maybe something like this:

```
CWWKC0651W: The MicroProfile configuration value for the key {0} has been used by an application before a checkpoint action.  The value of the key has changed since the checkpoint action completed on the server and the updated value might not be used by the application when the server process is restored.
```

Note that we say "might" because some applications may use something called [dynamic config](https://download.eclipse.org/microprofile/microprofile-config-2.0/microprofile-config-spec-2.0.html#_dynamic_configsource).  In that case the application will always get the most up to date value for the config key.

Hope that helps.